src/games/wows/api.ts

```diff
@@ -49,41 +49,45 @@ export const getWowsApiData = async (token: string, setGameApi: (api: Api) => vo
     const response: WowsShipsApiResponse = await axios.get(`${process.env.VUE_APP_MS_WG_API}/wows/encyclopedia/ships/?page_no=${i}`, { headers })
     apiShips = apiShips.concat(Object.values(response.data.data))
   }
-  let ships: Ship[] = apiShips.map((ship: ApiShipResponse) => {
-    return {
-      id: `${ship.ship_id}`,
-      name: ship.name,
-      tier: ship.tier,
-      type: ship.type as WowsShipType,
-      game: Game.WOWS,
-      image: gameInfo.data.data.ship_type_images[ship.type][ship.is_special ? 'image_elite' : ship.is_premium ? 'image_premium' : 'image'],
-      canvasImage: {
-        image: canvasIcons[ship.type as WowsShipType].image,
-        dimensions: {
-          width: canvasIcons[ship.type as WowsShipType].width,
-          height: canvasIcons[ship.type as WowsShipType].height
-        }
-      },
-      default: false,
-      data: {
-        artillery: ship.default_profile.artillery?.distance,
-        torpedo: ship.default_profile.torpedoes?.distance,
-        atbas: ship.default_profile.atbas?.distance,
-        concealmentPlane: ship.default_profile.concealment?.detect_distance_by_plane,
-        concealmentShip: ship.default_profile.concealment?.detect_distance_by_ship,
-        hydro: undefined,
-        radar: undefined
+  let ships: Ship[] = apiShips.map((ship: ApiShipResponse): Ship => ({
+    id: `${ship.ship_id}`,
+    uuid: uuid(),
```

Comment:
Why we need UUID generated here?

Reply:
Resolved.
The API returns an id for a particular ship but when we use the same ship twice we need to have a unique id for each one.
The solution will now involve a property apiId which is the id set by the API and then the existing id will become unique when multiple entities of the same type are created.